src/wp-includes/class-wp-term-query.php

```diff
@@ -1053,19 +1061,22 @@ protected function get_search_sql( $string ) {
 	 *
 	 * @since 4.9.8
 	 *
-	 * @param array $term_ids Term IDs.
+	 * @param array $terms_data Term data.
 	 * @return array
 	 */
-	protected function populate_terms( $term_ids ) {
+	protected function populate_terms( $terms_data ) {
```

Comment:
It would be good not to change the signature here (or allow an array of term ids too). `WP_Term_Query` isn't `final` so there's a risk of breaking classes extending this one for whatever reason.

A `@since 6.0.0 Accepts term data blah blah` will need to be added to the docblock.

Reply:
Fun fact, this method doesn't current accept an array of term ids, as documented. In most cases it is a object of term data. The documention and variable name is wrong. Changing this makes sense.

Reply:
Looks to me like it would, in fact, accept an array of term IDs, both before and after your changes (see the `else` clause below). However, @spacedmonkey is right that the variable name was incorrect - mea culpa, see https://core.trac.wordpress.org/changeset/43049 A change to `$terms_data` (or perhaps more simply `$terms`) with a note that this can either be an array of term objects OR term IDs seems appropriate.

Reply:
Updated the comment to be clearer and cleaner. Improving docs is a side bonus here!

Reply:
Changed name of param to terms.